kos-py/pykos/services/led_matrix.py

```diff
@@ -0,0 +1,96 @@
+"""LED Matrix service client."""
+
+from typing import NotRequired, TypedDict
+
+import grpc
+from google.protobuf.empty_pb2 import Empty
+
+from kos_protos import common_pb2, led_matrix_pb2, led_matrix_pb2_grpc
+
+
+class MatrixInfo(TypedDict):
+    """Information about the LED matrix.
+
+    Args:
+        width: Width in pixels
+        height: Height in pixels
+        brightness_levels: Number of brightness levels supported (1 for binary on/off)
+        color_capable: Whether the matrix supports color
+        bits_per_pixel: Number of bits used to represent each pixel
+        error: Optional error information
+    """
+
+    width: int
+    height: int
+    brightness_levels: int
+    color_capable: bool
+    bits_per_pixel: int
+    error: NotRequired[common_pb2.Error | None]
+
+
+class LEDMatrixServiceClient:
+    """Client for the LEDMatrixService.
+
+    This service allows controlling an LED matrix display.
+    """
+
+    def __init__(self, channel: grpc.Channel) -> None:
+        """Initialize the LED matrix service client.
+
+        Args:
+            channel: gRPC channel to use for communication.
+        """
+        self.stub = led_matrix_pb2_grpc.LEDMatrixServiceStub(channel)
+
+    def get_matrix_info(self) -> MatrixInfo:
+        """Get information about the LED matrix including dimensions and capabilities.
+
+        Returns:
+            MatrixInfo containing:
+                width: Width in pixels
+                height: Height in pixels
+                brightness_levels: Number of brightness levels supported
+                color_capable: Whether the matrix supports color
+                bits_per_pixel: Number of bits used to represent each pixel
+                error: Optional error information
+        """
+        response = self.stub.GetMatrixInfo(Empty())
+        return MatrixInfo(
+            width=response.width,
+            height=response.height,
+            brightness_levels=response.brightness_levels,
+            color_capable=response.color_capable,
+            bits_per_pixel=response.bits_per_pixel,
+            error=response.error if response.HasField("error") else None,
+        )
```

Comment:
so, i use the typed dictionaries to provide kwarg types for the typechecker - you don't need to return a typed dictionary here, you can just return the protobuf type, typechecker still works